OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUnitYoutubeVideoFragment.java

```diff
@@ -0,0 +1,436 @@
+        super.initTranscriptListView();
+        transcriptListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                final Caption currentCaption = transcriptAdapter.getItem(position);
+                if (currentCaption != null && youTubePlayer != null) {
+                    youTubePlayer.seekToMillis(currentCaption.start.getMseconds());
+                }
+            }
+        });
+    }
+
+    private void setVideoModel(){
+        videoModel = (DownloadEntry) DatabaseFactory.getInstance( DatabaseFactory.TYPE_DATABASE_NATIVE ).getVideoEntryByVideoId(unit.getId(), null);
+
+        if (videoModel == null) {
+            DownloadEntry e = new DownloadEntry();
+            e.videoId = unit.getId();
+            addVideoDatatoDb(e);
+            videoModel = e;
+        }
+    }
+
+    private  class StateChangeListener implements YouTubePlayer.PlayerStateChangeListener {
+        @Override
+        public void onLoading() {
+
+        }
+
+        @Override
+        public void onLoaded(String s) {
+
+        }
+
+        @Override
+        public void onAdStarted() {
+
+        }
+
+        @Override
+        public void onVideoStarted() {
+
+        }
+
+        @Override
+        public void onVideoEnded() {
+            youTubePlayer.seekToMillis(0);
+            youTubePlayer.pause();
+        }
+
+        @Override
+        public void onError(YouTubePlayer.ErrorReason errorReason) {
+            /**
+             * The most common errorReason is because there is a previous player running so this sets free it
+             * and reloads the fragment
+             */
+            if (attempts <= 3) {
+                if (previousYouTubePlayer != null){
+                    previousYouTubePlayer.release();
+                }
+                youTubePlayer.release();
+                initializeYoutubePlayer();
+                attempts++;
+            } else {
+                Toast.makeText(getActivity(), errorReason.toString(), Toast.LENGTH_LONG ).show();
+            }
+        }
+    }
+
+    private class PlaybackListener implements YouTubePlayer.PlaybackEventListener {
+
+        @Override
+        public void onPlaying() {
+            if (getActivity() != null && !fromLandscape) {
+                getActivity().setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_SENSOR);
+            }
+        }
+
+        @Override
+        public void onPaused() {
+
+        }
+
+        @Override
+        public void onStopped() {
+            if (wasHiding && getUserVisibleHint()) {
+                try {
+                    wasHiding = false;
+                    youTubePlayer.play();
+                } catch (Exception error) {
+                    initializeYoutubePlayer();
+                }
+
+            }
+            else if (!getUserVisibleHint()) {
```

Comment:
there is no need to release youtuebe player here, because of we are releasing the player in `setUserVisibleHint` method.